src/Core/Addon/Module/ModuleManagerBuilder.php

```diff
@@ -57,17 +57,25 @@ class ModuleManagerBuilder
     public static $translator = null;
     public static $addonsDataProvider = null;
     public static $categoriesProvider = null;
-    public static $instance = null;
+    public static $instance = array();
 
     /**
+     * @param null $baseUrl
+     *
      * @return ModuleManagerBuilder
      */
-    static public function getInstance() {
-        if (self::$instance == null) {
-            self::$instance = new self();
+    static public function getInstance($baseUrl = null) {
+        if ($baseUrl == null) {
+            $baseUrlKey = '';
+        } else {
+            $baseUrlKey = $baseUrl;
+        }
+
+        if (!array_key_exists($baseUrlKey, self::$instance)) {
+            self::$instance[$baseUrlKey] = new self($baseUrl);
         }
 
-        return self::$instance;
+        return self::$instance[$baseUrlKey];
```

Comment:
it shouldn't but it was just in case we want to use two different base_url in the same script :)
anyway, if we move the base_url code inside the class, it will not be needed anymore